File: godfather/factions/neutral/executioner.py
from godfather.game import Player
from godfather.factions.base import Faction


class ExecutionerNeutral(Faction):
    name = 'Executioner'
    id = 'neutral.executioner'
    win_con = 'Get your target lynched at any cost.'

    @property
    def category_name(self):
        return 'Neutral'

    def has_won_independent(self, player: Player):
        if player.is_alive and not player.target.is_alive and player.target.death_reason.startswith('lynched'):
            return True
        return False

File: src/lib/mafia/factions/neutral/Executioner.ts
import type Executioner from '@mafia/roles/neutral/Executioner';
import Faction from '@mafia/structures/Faction';
import type Player from '@mafia/structures/Player';
import { cast } from '@util/utils';

export default class ExecutionerFaction extends Faction {
	public name = 'Executioner';
	public winCondition = 'game/factions:executionerWinCondition';
	public independent = true;

	public hasWonIndependent(player: Player) {
		const { target } = cast<Executioner>(player.role);
		// a player has died if they are currently dead, or they were revived
		const isDead = !target.isAlive || target.flags.isRevived;
		// The exe was successful if the target was eliminated, however - retributionists clear
		// death reasons, so there's no way to check if the revived target was previously eliminated
		// If the target died at night, the Exe should become a Jester and never reach this code,
		// so we can assume that a revived target is a eliminated target
		const eliminationSuccessful = target.isAlive ? target.flags.isRevived : target.deathReason.includes('eliminated');
		return isDead && eliminationSuccessful;
	}
}
